packages/react-composites/src/composites/CallComposite/utils/MediaGalleryUtils.ts

```diff
@@ -0,0 +1,107 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { _formatString } from '@internal/acs-ui-common';
+import { RemoteParticipantState } from '@internal/calling-stateful-client';
+import { useEffect, useState } from 'react';
+import { useLocale } from '../../localization';
+import { useSelector } from '../hooks/useSelector';
+import { getRemoteParticipants } from '../selectors/baseSelectors';
+
+/**
+ * sets the announcement string whenever a Participant comes or goes from a call to be
+ * used by the system narrator.
+ *
+ * @returns string to be used by the narrator and Announcer component
+ *
+ * @internal
+ */
+export const useParticipantChangedAnnouncement = (): string => {
+  const locale = useLocale().strings.call;
+  const [announcerString, setAnnouncerString] = useState<string>('');
+  const remoteParticipantsObject = useSelector(getRemoteParticipants);
+  const remoteParticipants = Object.values(remoteParticipantsObject ?? {}).filter((p) => {
+    /**
+     * We need to make sure remote participants are in the connected state. If they are not
+     * they might not have their displayName set in the call state just yet.
+     */
+    if (p.state === 'Connected') {
+      return p;
+    }
+    return;
+  });
+  const [currentParticipants, setCurrentParticipants] = useState<RemoteParticipantState[]>(remoteParticipants);
+
+  useEffect(() => {
+    const setParticipantEventString = (string: string): void => {
+      setAnnouncerString('');
+      setAnnouncerString(string);
+    };
+    if (remoteParticipants.length < currentParticipants.length) {
+      //someone left
+      const whoLeft = currentParticipants.filter((p) => {
+        if (!remoteParticipants.includes(p)) {
```

Comment:
I'm a bit wary of leaving behind hidden O(N*N) cost in the codebase, even when we think that N will stay small.

I'd recommend creating sets for `currentParticipants` and `remoteParticipants` first so that you are assured the repeated `.includes` test doesn't blow up.

Funsies:

![image](https://user-images.githubusercontent.com/82062616/202032541-98a2d8c9-cfbd-4a45-906a-9b8bf7de8fe6.png)
https://www.joelonsoftware.com/2001/12/11/back-to-basics/